src/libraries/System.Private.CoreLib/src/System/Collections/Generic/ValueListBuilder.cs

```diff
@@ -84,7 +84,7 @@ public void Dispose()
 
         private void Grow()
         {
-            T[] array = ArrayPool<T>.Shared.Rent(_span.Length * 2);
+            T[] array = ArrayPool<T>.Shared.Rent(Math.Max(_span.Length + 1, _span.Length * 2));
```

Comment:
```suggestion
            T[] array = ArrayPool<T>.Shared.Rent(Math.Max(1, _span.Length * 2));
```

?

(for me it's easier to understand.  up to you :)).

Reply:
The +1 version also handles more overflow coditions.

Reply:
possibly, though i imagine then you're in a world of pain, as you'll keep growing by one and then copying eveyrthing over.  up to you :D.